test/test.tools.js

```diff
@@ -50,37 +60,29 @@ describe("Monkeypatching", () => {
 
 describe("Test tools for parsing new gmail body_params", () => {
     const gmail = new Gmail();
-    const data =  JSON.parse(testData.new_gmail_archive_action_body_params);
-
-    it("get thread id", () => {
-        const thread = gmail.tools.get_thread_id(data);
 
-        assert.equal(thread, 'thread-f:1600724307680265309');
+    it("parses archived messages", () => {
+        const xhrData = testData.new_gmail_archive_action_body_params;
+        testXhrEventParsing(xhrData, "archive");
     });
-    it("get thread data", () => {
-        const mockThreadData = data[2][7];
-        const threadData = gmail.tools.get_thread_data(data);
 
-        assert.deepEqual(threadData, mockThreadData);
+    it("parses deleted messages", () => {
+        const xhrData = testData.new_gmail_delete_action_body_params;
+        testXhrEventParsing(xhrData, "delete");
     });
-    it("get messages ids", () => {
-        const mockMessageIds = ['msg-f:1600724307680265309', 'msg-f:1600724938213937205', 'msg-f:1600725174437456906', 'msg-f:1600725319255992336', 'msg-f:1600725448529658711'];
-        const threadData = gmail.tools.get_thread_data(data);
```

Comment:
And this one.

We just want to replace the tests which concretely cover action/events, and not other thread/message meta-data.